workers/unity/Packages/com.improbable.gdk.testutils/WorkerOpFactory.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            var schemaRequest = new SchemaCommandRequest(componentId, commandIndex);
+            var op = new CommandRequestOp
+            {
+                Request = new CommandRequest(schemaRequest),
+                RequestId = new RequestId<IncomingCommandRequest>(requestId)
+            };
+
+            return new WrappedOp<CommandRequestOp>(op);
+        }
+
+        public static WrappedOp<CommandResponseOp> CreateCommandResponseOp(uint componentId, uint commandIndex, long requestId)
+        {
+            var schemaResponse = new SchemaCommandResponse(componentId, commandIndex);
+
+            var op = new CommandResponseOp
+            {
+                Response = new CommandResponse(schemaResponse),
+                RequestId = new RequestId<OutgoingCommandRequest>(requestId)
+            };
+
+            return new WrappedOp<CommandResponseOp>(op);
+        }
+
+        public static WrappedOp<DisconnectOp> CreateDisconnectOp(string reason)
+        {
+            var op = new DisconnectOp { Reason = reason };
+            return new WrappedOp<DisconnectOp>(op);
+        }
+
+        public static WrappedOp<CreateEntityResponseOp> CreateCreateEntityResponseOp(long requestId)
+        {
+            var op = new CreateEntityResponseOp
+            {
+                RequestId = new RequestId<CreateEntityRequest>(requestId)
+            };
+
+            return new WrappedOp<CreateEntityResponseOp>(op);
+        }
+
+        public static WrappedOp<DeleteEntityResponseOp> CreateDeleteEntityResponseOp(long requestId)
+        {
+            var op = new DeleteEntityResponseOp
+            {
+                RequestId = new RequestId<DeleteEntityRequest>(requestId)
+            };
+
+            return new WrappedOp<DeleteEntityResponseOp>(op);
+        }
+
+        public static WrappedOp<ReserveEntityIdsResponseOp> CreateReserveEntityIdsResponseOp(long requestId)
+        {
+            var op = new ReserveEntityIdsResponseOp
+            {
+                RequestId = new RequestId<ReserveEntityIdsRequest>(requestId)
+            };
+
+            return new WrappedOp<ReserveEntityIdsResponseOp>(op);
+        }
+
+        public static WrappedOp<EntityQueryResponseOp> CreateEntityQueryResponseOp(long requestId)
+        {
+            var op = new EntityQueryResponseOp
+            {
+                RequestId = new RequestId<EntityQueryRequest>(requestId)
+            };
+
+            return new WrappedOp<EntityQueryResponseOp>(op);
+        }
+    }
+
+    /// <summary>
+    ///     A wrapped Worker Op that implements Dispose so that the allocated native memory is properly freed.
+    ///     Ensure to call Dipose or use WrappedOp with a using scope.
+    /// </summary>
+    /// <typeparam name="T">The worker op type. E.g. - AddComponentOp</typeparam>
+    public struct WrappedOp<T> : IDisposable where T : struct
+    {
+        public T Op;
+
+        public WrappedOp(T op)
+        {
+            Op = op;
```

Comment:
should we add a check here to ensure you only create a `WrappedOp` with a valid op type? or make it `private` and you always have to use the factory to get a `WrappedOp` object

Reply:
++ private constructor